docs/high-availability/provision.sh

```diff
@@ -0,0 +1,180 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+set -x
+
+# Grep the whole IP because otherwise sometimes vagrant attaches extra mystery dynamic IPs to eth1.
+IP=`ip -o addr | grep '192.168.4' | cut -d' ' -f 7 | cut -d'/' -f 1`
+
+echo "Using IP $IP for this machine."
+
+function initialize {
+    systemctl disable iptables-services firewalld
+    echo "disabling selinux"
+    (setenforce 0 || echo "selinux might already be disabled...")
+    yum install -y docker
+
+    ### Important : The kube version MUST match the containers in the manifests.
+    ### Otherwise lots of API errors.
+    yum install -y http://cbs.centos.org/kojifiles/packages/kubernetes/0.17.1/3.el7/x86_64/kubernetes-node-0.17.1-3.el7.x86_64.rpm
+    mkdir -p -m 777 /etc/kubernetes/manifests
+    ### just to make it easy to hack around as non root user
+    groupadd docker
+    gpasswd -a vagrant docker
+    systemctl restart docker
+}
+
+function start_kubelet {
+    systemctl enable docker
+    ### We need a custom unit file with the --config/ option
+    cp /vagrant/etc_kubernetes_kubelet /etc/kubernetes/kubelet
+    systemctl enable kubelet
+    ### Not sure why, but this restart is required?
+    sleep 2
+    systemctl restart kubelet
+}
+
+### Not the best idea if using flannel.  Because of the circular dependency.
+function write_etcd_manifest {
+
+        ### I know this looks fancy, but
+        ### Basically, this is just setting up ETCD config file w/ IP Addresses
+         cat /vagrant/etcd.manifest | \
+        sed "s/NODE_NAME/`hostname`/g" | \
+        sed "s/NODE_IP/$IP/g" > /etc/kubernetes/manifests/etcd.manifest
+}
+
+### Test of ETCD Members.
+
+function test_etcd {
+
+    echo "----------- DEBUG ------------ KUBELET LOGS -----------------"
+    ( journalctl -u kubelet | grep -A 20 -B 20 Fail || echo "no failure in logs")
+    echo "----------- END DEBUG OF KUBELET ----------------------------"
+
+    ( curl http://kube0.ha:2379 > /tmp/curl_output || echo "failed etcd!!!" )
+    if [ -s /tmp/curl_output ]; then
+         echo "etcd success"
+    else
+         echo "etcd failure.  exit!"
+         exit 100
+    fi
+
+}
+
+function k8petstore {
+     ### run K8petstore .  Should work perfectly IFF flannel and so on is setup properly.
+    wget https://raw.githubusercontent.com/GoogleCloudPlatform/kubernetes/release-0.17/examples/k8petstore/k8petstore.sh
```

Comment:
only because Its specifically tested to work with this stuff.  
As you mentioned, we'll update to 1x and then we can use the Node based implementation ( https://github.com/GoogleCloudPlatform/kubernetes/blob/master/examples/k8petstore/k8petstore-nodeport.sh ).